atlasdb-tests-shared/src/test/java/com/palantir/atlasdb/transaction/impl/SnapshotTransactionTest.java

```diff
@@ -1178,6 +1181,174 @@ public void cleanup() {}
+        List<Cell> cells = serializableTxManager.runTaskWithRetry(tx -> {
+            tx.put(TABLE, ImmutableMap.of(inBetweenCell, value));
+            return Lists.transform(
+                    Lists.newArrayList(tx.getRowsColumnRange(
+                            TABLE, ImmutableList.of(row), new ColumnRangeSelection(null, null), 10)),
+                    Map.Entry::getKey);
+        });
+        assertThat(cells).containsExactly(firstCell, inBetweenCell, secondCell);
+    }
+
+    @Test
+    public void getRowsColumnRangeIncludesLocalWritesWhenRowEntirelyAbsentFromKvs() {
+        byte[] row = "foo".getBytes(StandardCharsets.UTF_8);
+        Cell beforeRange = Cell.create(row, "james".getBytes(StandardCharsets.UTF_8));
+        Cell inRangeOne = Cell.create(row, "jeremy".getBytes(StandardCharsets.UTF_8));
+        Cell inRangeTwo = Cell.create(row, "tom".getBytes(StandardCharsets.UTF_8));
+        Cell afterRange = Cell.create(row, "will".getBytes(StandardCharsets.UTF_8));
+        byte[] value = new byte[1];
+
+        List<Cell> cells = serializableTxManager.runTaskWithRetry(tx -> {
+            tx.put(TABLE, ImmutableMap.of(beforeRange, value, inRangeOne, value, inRangeTwo, value, afterRange, value));
+            return Lists.transform(
+                    Lists.newArrayList(tx.getRowsColumnRange(
+                            TABLE,
+                            ImmutableList.of(row),
+                            new ColumnRangeSelection(
+                                    RangeRequests.nextLexicographicName(beforeRange.getColumnName()),
+                                    RangeRequests.previousLexicographicName(afterRange.getColumnName())),
+                            10)),
+                    Map.Entry::getKey);
+        });
+        assertThat(cells).containsExactly(inRangeOne, inRangeTwo);
+    }
+
+    @Test
+    public void getRowsColumnRangeIncludesIntermediateRowsWithOnlyLocalWritesCorrectly() {
+        byte[] row1 = "apple".getBytes(StandardCharsets.UTF_8);
+        byte[] row2 = "banana".getBytes(StandardCharsets.UTF_8);
+        byte[] row3 = "cherry".getBytes(StandardCharsets.UTF_8);
+        byte[] value = new byte[1];
+
+        Cell firstCell = Cell.create(row1, "apricot".getBytes(StandardCharsets.UTF_8));
+        Cell secondCell = Cell.create(row2, "bamboo".getBytes(StandardCharsets.UTF_8));
+        Cell thirdCell = Cell.create(row3, "coconut".getBytes(StandardCharsets.UTF_8));
+
+        serializableTxManager.runTaskWithRetry(tx -> {
+            tx.put(TABLE, ImmutableMap.of(firstCell, value, thirdCell, value));
+            return null;
+        });
+
+        List<Cell> cells = serializableTxManager.runTaskWithRetry(tx -> {
+            tx.put(TABLE, ImmutableMap.of(secondCell, value));
+            return Lists.transform(
+                    Lists.newArrayList(tx.getRowsColumnRange(
+                            TABLE, ImmutableList.of(row1, row2, row3), new ColumnRangeSelection(null, null), 10)),
```

Comment:
given by comment about order earlier, may be good to change the order of rows here? Or maybe even create a separate test for it?

Reply:
Yeah I added another test that inserts another row and queries the remainder in a different order. I thought prefix/suffix/intermediate should cover most orderings, though it's fair to have a few more cases for "intermediate"